01-cloudformation/Lesson-1-3/cdk.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import boto3
+import json
+import sys
+import os.path
+
+
+        
+def stack_exists(Client, Stack_Name, Template_File): 
+    # Check to see if that stack exists in the region
+    try:
+        if Client.describe_stacks(
+            StackName=Stack_Name):
+            return True
+    except:
+            return False
+
+def stack_update(Client, Stack_Name, Template_File):
+    print("Updating {} stack in region {}".format(Stack_Name, Client.meta.region_name))
+    response = Client.update_stack(
+        StackName = Stack_Name,
+        TemplateBody = Template_File,
+        Parameters = [{
+                'ParameterKey': 'BucketNameParam',
+                'ParameterValue': 'mmcelreath-cdk-update' 
+            }]
+    )
+    print(response)
+
+
+
+def stack_create(Client, Stack_Name, Template_File):
+    print("Creating {} stack in region {}".format(Stack_Name, Client.meta.region_name))
+    response = Client.create_stack(
+        StackName = Stack_Name,
+        TemplateBody = Template_File,
+        Parameters = [{
+                'ParameterKey': 'BucketNameParam',
+                'ParameterValue': 'mmcelreath-cdk' 
+            }]
+    )
+    print(response)
+
+def stack_delete(Client, Stack_Name):
+    print("Deleting {} stack in region {}".format(Stack_Name, Client.meta.region_name))
+    response = Client.delete_stack(
+		StackName=Stack_Name
+	)
+    print(response)
+
+
+
+
+def main():
+    
+    # Create the template body object since we'll need it everywhere.
+    Template_File = open("s3.yaml", "r").read()
+    # Read the regions file and delcare an object
+    Regions_File = open("regions.json")
+    Data = json.load(Regions_File)
+
+
+    # Parse out the regions and create a Client for each region. 
+    for REGIONS in Data["Regions"]:
+        Client = boto3.client('cloudformation', region_name=REGIONS["RegionName"])
+        try:
+            if len(sys.argv[3]) >3 and sys.argv[3] == "delete":
```

Comment:
I would almost always use argparse for scripts like this unless they are super super simple